app/src/test/java/com/group80/uoftinder/GenerateCompatibilityListUnitTest.java

```diff
@@ -0,0 +1,135 @@
+package com.group80.uoftinder;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import com.group80.uoftinder.create_account_use_case.CreateAccountInteractor;
+import com.group80.uoftinder.entities.User;
+import com.group80.uoftinder.feed.GenerateCompatibilityList;
+import com.group80.uoftinder.feed.RecommendationPresenter;
+import com.group80.uoftinder.feed.RecommendationView;
+import com.group80.uoftinder.feed.UserScoreFacade;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * A GenerateCompatibilityListUnitTest class that tests the functionality of the
+ * GenerateCompatibilityList class
+ */
+public class GenerateCompatibilityListUnitTest {
+
+    /**
+     * Test that GenerateCompatibilityList.orderCompatibilityList reorders the
+     * compatibility list from most compatible user to least compatible user when there are
+     * 3 total users
+     */
+    @Test
+    public void orderCompatibilityListTest1() {
+        User curUser = new User("curUser");
+        int curUserScore = 11682;
+        curUser.setScore(curUserScore);
+//        genCompatibilityList.setCurUser(curUser);
+
+        CreateAccountInteractor.setAnswerSchema(new int[] {3, 4, 4, 5, 4});
+        CreateAccountInteractor.setIsMultiSelect(new boolean[] {false, true, true, false, false});
+
+        List<List<Integer>> userAnswers = new ArrayList<>();
+        userAnswers.add(Collections.singletonList(1));
+        userAnswers.add(Arrays.asList(1, 2));
+        userAnswers.add(Arrays.asList(0, 1, 3));
+        userAnswers.add(Collections.singletonList(0));
+        userAnswers.add(Collections.singletonList(2));
+        curUser.setAnswers(userAnswers);
+
+        GenerateCompatibilityList genCompatibilityList = new GenerateCompatibilityList(curUser);
+        User user2 = new User("user2");
+        User user3 = new User("user3");
+
+        int user2Score = 18642;
+        int user3Score = 3891;
+
+        user2.setScore(user2Score);
+        user3.setScore(user3Score);
+
+        List<User> initCompList = new ArrayList<>(Arrays.asList(user2, user3));
```

Comment:
Eventually, might be good to test with more users! Maybe it would be good to make a workflow that randomly generates Users with for a given answer schema and spits out their userAnswers list. Then, we can calculate their score through the ```UserScoreFacade``` class and test!